x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -449,51 +431,38 @@ public synchronized AllowedRealmType allowedRealmType() {
     /**
      * @return whether custom role providers are allowed based on the license {@link OperationMode}
      */
-    public synchronized boolean isCustomRoleProvidersAllowed() {
-        final boolean isSecurityCurrentlyEnabled =
-            isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
-        return isSecurityCurrentlyEnabled && (status.mode == OperationMode.PLATINUM || status.mode == OperationMode.TRIAL)
-                && status.active;
+    public boolean isCustomRoleProvidersAllowed() {
+        return isAllowedByLicenseAndSecurity(OperationMode.PLATINUM, true, true, true);
     }
 
     /**
      * @return whether the Elasticsearch {@code TokenService} is allowed based on the license {@link OperationMode}
      */
-    public synchronized boolean isTokenServiceAllowed() {
-        final OperationMode mode = status.mode;
-        final boolean isSecurityCurrentlyEnabled = isSecurityEnabled(mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
-        return isSecurityCurrentlyEnabled && (mode == OperationMode.GOLD || mode == OperationMode.PLATINUM || mode == OperationMode.TRIAL);
+    public boolean isTokenServiceAllowed() {
+        return isAllowedByLicenseAndSecurity(OperationMode.GOLD, true, false, true);
     }
 
     /**
      * @return whether the Elasticsearch {@code ApiKeyService} is allowed based on the current node/cluster state
      */
-    public synchronized boolean isApiKeyServiceAllowed() {
-        final OperationMode mode = status.mode;
-        final boolean isSecurityCurrentlyEnabled = isSecurityEnabled(mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
-        return isSecurityCurrentlyEnabled;
+    public boolean isApiKeyServiceAllowed() {
+        return isAllowedBySecurity();
```

Comment:
This reads strangely to me. 
I realise that, in terms of readability, it's logically equivalent to the `isAllowedByLicenseAndSecurity` method, but here with no parameters, it just looks like a mistake.

Given this is the only place it's called, I would just replace it with `isAllowedByLicenseAndSecurity(OperationMode.BASIC, true, false, true);`